src/App.tsx

```diff
@@ -14,11 +17,105 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selected: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selected: [],
+  };
+
+  selectItem = (item: string) => {
+    this.setState((state) => ({
+      selected: [...state.selected, item],
+    }));
+  };
+
+  removeItem = (item: string) => {
+    this.setState((state) => ({
+      selected: state.selected.filter(el => el !== item),
+    }));
+  };
+
+  removeItems = () => {
+    this.setState({ selected: [] });
+  };
+
+  getItems = () => {
+    const { selected } = this.state;
+
+    if (selected.length === 1) {
+      return `${selected.join()} is selected`;
+    }
+
+    return `${selected.join(', ')} are selected`;
+  };
+
+  render() {
+    const { selected } = this.state;
+
+    return (
+      <div className="box">
+        <h1 className="title">
+          {selected.length !== 0
+            ? this.getItems()
+            : 'No goods selected'}
+          {' '}
+          {selected.length !== 0 && (
+            <button
+              type="button"
+              className="button is-danger"
+              onClick={this.removeItems}
+            >
+              X
+            </button>
+          )}
+        </h1>
+
+        <div>
+          <ul className="content">
+            {goodsFromServer.map(good => (
+              <li key={good}>
+                <div className="block">
+                  <span className={classNames(
+                    {
+                      'tag is-success is-light': selected.includes(good),
+                    },
+                  )}
+                  >
+                    {good}
+                  </span>
+                  {' '}
+                  {selected.includes(good)
```

Comment:
run includes method once, not twice

Reply:
do I need to write a new method and then run it 2 times? or where I need to save result of includes, in state?